src/Plugin.cpp

```diff
@@ -73,134 +77,214 @@ void Plugin::init() {
-      VistaOpenSGMaterialTools::SetSortKeyOnSubtree(
-          deepSpaceDotNode, static_cast<int>(cs::utils::DrawOrder::eTransparentItems) - 1);
-      mDeepSpaceDotNodes.emplace_back(deepSpaceDotNode);
-
       // do not perform distance culling for DeepSpaceDots
       dot->pVisibleRadius = -1;
+      dot->pVisible       = true;
 
-      if (settings.second.mTrail) {
-        // hide dot as soon as the corresponding trajectory is hidden
-        dot->pVisible.connectFrom(mTrajectories.back()->pVisible);
-      } else {
-        dot->pVisible = true;
-      }
-
-      mDeepSpaceDots.push_back(dot);
+      mDeepSpaceDots[settings.first] = dot;
     }
   }
 
-  mGuiManager->addSettingsSectionToSideBarFromHTML("Trajectories", "radio_button_unchecked",
-      "../share/resources/gui/trajectories-settings.html");
-
-  mGuiManager->getGui()->registerCallback("trajectories.setEnableTrajectories",
-      "Enables or disables the rendering of trajectories.",
-      std::function([this](bool value) { mProperties->mEnableTrajectories = value; }));
+  // For the trajectories we try to re-use as many as possible as they are quite expensive to
+  // construct. First try to re-configure existing trajectories. A trajectory is re-used if it
+  // shares the same target anchor name.
+  auto trajectory = mTrajectories.begin();
+  while (trajectory != mTrajectories.end()) {
+    auto settings = mPluginSettings->mTrajectories.find(trajectory->first);
+
+    // If there are settings for this trajectory, reconfigure it.
+    if (settings != mPluginSettings->mTrajectories.end() && settings->second.mTrail) {
+      auto parentAnchor = mAllSettings->mAnchors.find(settings->second.mTrail->mParent);
+      auto targetAnchor = mAllSettings->mAnchors.find(settings->first);
+
+      // Ignore wrongly configured trajectories for now. The error will be emitted when we try to
+      // add this as a new trajectory.
+      if (parentAnchor != mAllSettings->mAnchors.end() &&
+          targetAnchor != mAllSettings->mAnchors.end()) {
+
+        auto [parentStartExistence, parentEndExistence] = parentAnchor->second.getExistence();
+        auto [targetStartExistence, targetEndExistence] = targetAnchor->second.getExistence();
+        trajectory->second->setStartExistence(std::max(parentStartExistence, targetStartExistence));
+        trajectory->second->setEndExistence(std::min(parentEndExistence, targetEndExistence));
+        trajectory->second->setCenterName(parentAnchor->second.mCenter);
+        trajectory->second->setFrameName(parentAnchor->second.mFrame);
+        trajectory->second->setTargetCenterName(targetAnchor->second.mCenter);
+        trajectory->second->setTargetFrameName(targetAnchor->second.mFrame);
+        trajectory->second->pSamples = settings->second.mTrail->mSamples;
+        trajectory->second->pLength  = settings->second.mTrail->mLength;
+        trajectory->second->pColor   = VistaColor(
```

Comment:
Why are we using `VistaColor` over a `glm::vec`? This would be more consistent across the code base and avoids conversions.